airflow/ui/src/pages/Dashboard/Stats/DAGImportErrorsModal.tsx

```diff
@@ -0,0 +1,125 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Heading, Text, HStack, Input } from "@chakra-ui/react";
+import { useEffect, useState } from "react";
+import { PiFilePy } from "react-icons/pi";
+
+import type { ImportErrorResponse } from "openapi/requests/types.gen";
+import Time from "src/components/Time";
+import { Accordion, Dialog } from "src/components/ui";
+import { Pagination } from "src/components/ui/Pagination";
+
+type ImportDAGErrorModalProps = {
+  importErrors: Array<ImportErrorResponse>;
+  onClose: () => void;
+  open: boolean;
+};
+
+const PAGE_LIMIT = 5;
+
+export const DAGImportErrorsModal: React.FC<ImportDAGErrorModalProps> = ({
+  importErrors,
+  onClose,
+  open,
+}) => {
+  const [page, setPage] = useState(1);
+  const [searchQuery, setSearchQuery] = useState("");
+  const [filteredErrors, setFilteredErrors] = useState(importErrors);
+
+  const startRange = (page - 1) * PAGE_LIMIT;
+  const endRange = startRange + PAGE_LIMIT;
+  const visibleItems = filteredErrors.slice(startRange, endRange);
+
+  const onOpenChange = () => {
+    setSearchQuery("");
+    setPage(1);
+    onClose();
+  };
+
+  useEffect(() => {
+    const query = searchQuery.toLowerCase();
+    const filtered = importErrors.filter((error) =>
+      error.filename.toLowerCase().includes(query),
+    );
+
+    setFilteredErrors(filtered);
+    setPage(1);
+  }, [searchQuery, importErrors]);
+
+  return (
+    <Dialog.Root
+      onOpenChange={onOpenChange}
+      open={open}
+      scrollBehavior="inside"
+      size="xl"
+    >
+      <Dialog.Content backdrop>
+        <Dialog.Header>
+          <Heading size="xl">Dag Import Errors</Heading>
+          <Input
+            mt={4}
+            onChange={(letters) => setSearchQuery(letters.target.value)}
```

Comment:
```suggestion
            onChange={(event) => setSearchQuery(event.target.value)}
```

We're accessing a ChangeEvent. Let's not rename it.